js/src/components/google-combo-account-card/connect-ads/connect-ads.js

```diff
@@ -1,122 +1,54 @@
 /**
  * External dependencies
  */
-import { useEffect, useState } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import useApiFetchCallback from '.~/hooks/useApiFetchCallback';
-import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
-import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
-import { useAppDispatch } from '.~/data';
-import useGoogleAdsAccountReady from '.~/hooks/useGoogleAdsAccountReady';
 import AccountCard from '.~/components/account-card';
-import AdsAccountSelectControl from '.~/components/ads-account-select-control';
-import ConnectedIconLabel from '.~/components/connected-icon-label';
-import ConnectAdsFooter from './connect-ads-footer';
-import LoadingLabel from '.~/components/loading-label';
-import ConnectButton from '.~/components/google-ads-account-card/connect-ads/connect-button';
+import ConfirmCreateModal from './confirm-create-modal';
+import ConnectExistingAccount from './connect-existing-account';
+import UpsertingAccount from './upserting-account';
 
 /**
  * ConnectAds component renders an account card to connect to an existing Google Ads account.
  *
+ * @param {Object} props Component props.
+ * @param {Function} props.onRequestCreate A callback to fire when creating a new account.
+ * @param {string} props.upsertingAction The action the user is performing. Either 'create' or 'update'.
```

Comment:
Continue on https://github.com/woocommerce/google-listings-and-ads/pull/2651#discussion_r1837998702.

The `null` type is missing.

```suggestion
 * @param {string|null} props.upsertingAction The action the user is performing. Possible values are 'create', 'update', or null.
```

Reply:
Thanks @eason9487 . JSDoc updated.